Akismet/AkismetReal.php

```diff
@@ -83,6 +83,25 @@ public function isSpam(array $data)
         }
     }
 
+    function submitSpam(array $data)
+    {
+        $fullData = array_merge($this->getRequestData(), $data);
+
+        if ($this->throwExceptions) {
+            return $this->adapter->submitSpam($fullData);
```

Comment:
Instead of having this `return submitSpam` in this method twice (which confused me for a bit), how about deleting this `if` block, then putting an `if (throwExceptions) { throw $e) }` at the beginning of the `catch` block instead? That way you have one call to `submitSpam`, then if an exception does occur, you decide in the `catch` block what to do with it. That seems cleaner and clearer to me.